services/tables/src/main/java/com/linkedin/openhouse/tables/toggle/repository/ToggleStatusesRepositoryImpl.java

```diff
@@ -1,29 +1,48 @@
 package com.linkedin.openhouse.tables.toggle.repository;
 
+import com.linkedin.openhouse.housetables.client.api.ToggleStatusApi;
+import com.linkedin.openhouse.housetables.client.model.EntityResponseBodyToggleStatus;
+import com.linkedin.openhouse.tables.toggle.ToggleStatusMapper;
 import com.linkedin.openhouse.tables.toggle.model.TableToggleStatus;
 import com.linkedin.openhouse.tables.toggle.model.ToggleStatusKey;
 import java.util.Optional;
-import org.springframework.stereotype.Component;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
 /**
- * THIS IS A TEMPORARY PLACEHOLDER, without this the whole springboot application is failed to start
- * given missing injection entity for {@link ToggleStatusesRepository}.
+ * A base implementation for {@link ToggleStatusesRepository} that represents an interface for fetch
+ * feature-toggle-status of a table entity.
  */
-@Component
-public class BaseToggleStatusesRepository implements ToggleStatusesRepository {
+@Repository
+@Slf4j
+public class ToggleStatusesRepositoryImpl implements ToggleStatusesRepository {
+  @Autowired private ToggleStatusApi apiInstance;
+
+  @Autowired private ToggleStatusMapper toggleStatusMapper;
+
   @Override
-  public <S extends TableToggleStatus> S save(S entity) {
-    throw new UnsupportedOperationException("Not supported");
+  public Optional<TableToggleStatus> findById(ToggleStatusKey toggleStatusKey) {
+    return apiInstance
+        .getTableToggleStatus(
+            toggleStatusKey.getDatabaseId(),
+            toggleStatusKey.getTableId(),
+            toggleStatusKey.getFeatureId())
+        .map(EntityResponseBodyToggleStatus::getEntity)
+        .map(s -> toggleStatusMapper.toTableToggleStatus(toggleStatusKey, s))
+        .blockOptional();
   }
 
   @Override
-  public <S extends TableToggleStatus> Iterable<S> saveAll(Iterable<S> entities) {
-    throw new UnsupportedOperationException("Not supported");
+  public <S extends TableToggleStatus> S save(S entity) {
+    throw new UnsupportedOperationException(
+        "Write Operation into Toggle status API is not supported");
   }
 
   @Override
-  public Optional<TableToggleStatus> findById(ToggleStatusKey toggleStatusKey) {
-    return Optional.empty();
+  public <S extends TableToggleStatus> Iterable<S> saveAll(Iterable<S> entities) {
+    throw new UnsupportedOperationException(
+        "Write Operation into Toggle status API is not supported");
   }
 
   @Override
```

Comment:
can we throw unsupportedException for other methods as well like existsByID, findall etc?

Reply:
addressed.